packages/wallet-sdk/src/sign/scw/SCWSigner.ts

```diff
@@ -175,9 +166,42 @@ export class SCWSigner implements Signer {
     const response = await this.sendEncryptedRequest(request);
     const decrypted = await this.decryptResponseMessage(response);
 
+    return this.handleResponse(request, decrypted);
+  }
+
+  private async handleResponse(request: RequestArguments, decrypted: RPCResponse) {
     const result = decrypted.result;
     if ('error' in result) throw result.error;
 
+    switch (request.method) {
+      case 'eth_requestAccounts': {
+        const accounts = result.value as Address[];
+        this.accounts = accounts;
+        this.storage.storeObject(ACCOUNTS_KEY, accounts);
+        this.callback?.('accountsChanged', accounts);
+        break;
+      }
+      case 'wallet_connect': {
+        const response = result.value as WalletConnectResponse;
+        const accounts = response.accounts.map((account) => account.address);
+        this.accounts = accounts;
+        this.storage.storeObject(ACCOUNTS_KEY, accounts);
+        this.callback?.('accountsChanged', accounts);
```

Comment:
hmmm wonder if we should return sub account address here

Reply:
hmm... yeah I think so. We do that in other places